packages/block-library/src/cover/test/edit.native.js

```diff
@@ -541,7 +540,7 @@ describe( 'color settings', () => {
 	} );
 
 	it( 'displays the hex color value in the custom color picker', async () => {
-		HsvColorPicker.mockImplementation( ( props ) => {
+		ColorPicker.mockImplementation( ( props ) => {
```

Comment:
In context of [this comment](https://github.com/WordPress/gutenberg/pull/53329#discussion_r1291507353), I've updated this Cover block test to remove the outdated `HsvColorPicker` mock. However, in the process of digesting the (now removed) Color Picker test in favor of this one, I think I've confused myself on the best place to implement the `testID="color-picker"` param that is being queried [here](https://github.com/WordPress/gutenberg/blob/20ae378b7065bf75b2af6809b179dfa38028eb4b/packages/block-library/src/cover/test/edit.native.js#L564). In best practice, should this be added to the [Color Picker params itself](https://github.com/WordPress/gutenberg/blob/59a346da13c8c0d759bfe9609c15afb737922e53/packages/components/src/color-picker/index.native.js#L123-L157), or to the [Cover block implementation](https://github.com/WordPress/gutenberg/blob/59a346da13c8c0d759bfe9609c15afb737922e53/packages/block-library/src/cover/edit.native.js#L391-L411) of the Color Picker?

Reply:
My high level answer would be to keep it simple. Place the test ID directly on the element you need to query (the element handling the event callback), rather than attempting to pass it down from a parent or test context. 

Passing down could make the the components more flexible, but avoid that complexity until we need it.

Reply:
Thanks. I ended up placing it on the [hsv-color-picker component](https://github.com/WordPress/gutenberg/pull/53329/commits/d14b5f62af9c4d74e6c3e04be5962abac1fbc510) itself.